eclipse-collections-code-generator/src/main/resources/api/map/mutablePrimitiveObjectMap.stg

```diff
@@ -45,20 +53,71 @@ public interface Mutable<name>ObjectMap\<V> extends <name>ObjectMap\<V>, Mutable
     }
 
     /**
+     * Puts all of the key/value mappings from the specified map into this map. If this
+     * map already has a value associated with one of the keys in the map, it will be
+     * replaced with the value in {@code map}.
+     * @param map the map to copy into this map
      * @since 5.0.
      */
     void putAll(<name>ObjectMap\<? extends V> map);
 
+    /**
+     * Removes the mapping associated with the key, if one exists, from the map.
+     * @param key the key to remove
+     * @see #remove(<type>)
+     */
     V removeKey(<type> key);
 
+    /**
+     * Removes the mapping associated with the key, if one exists, from the map.
+     * @param key the key to remove
+     * @see #removeKey(<type>)
+     */
     V remove(<type> key);
 
+    /**
+     * Retrieves the value associated with the key if one exists; if it does not,
+     * associates a value with the key.
+     * a new value with they key
+     * @param key the key
+     * @param value the value to associate with {@code key} if no such mapping exists
+     * @return the value associated with key, if one exists, or {@code value} if not
+     */
     V getIfAbsentPut(<type> key, V value);
 
+    /**
+     * Retrieves the value associated with the key if one exists; if it does not,
+     * associates the result of invoking the value supplier with the key.
+     * @param key the key
+     * @param function the supplier that provides the value if no mapping exists for {@code key}
+     * @return the value associated with the key, if one exists, or the result of
+     * invoking {@code function} if not
+     */
     V getIfAbsentPut(<type> key, Function0\<? extends V> function);
 
+    /**
+     * Retrieves the value associated with the key if one exists; if it does not,
+     * associates the result of invoking the value function with the key.
+     * @param key the key
+     * @param function the function that provides the value if no mapping exists.
+     * The {@code key} will be passed as the argument to the function.
+     * @return the value associated with the key, if one exists, or the result of
+     * invoking {@code function} with {@code key} if not
+     */
     V getIfAbsentPutWithKey(<type> key, <name>ToObjectFunction\<? extends V> function);
 
+    /**
+     * Retrieves the value associated with the key if one exists; if it does not,
+     * associates the result of invoking the value function with the key.
+     * @param key the key
```

Comment:
The value function is invoked with the additional parameter here, not the key.

Reply:
P.S. I feel like those parameters/methods overloads are largely unnecessary because a function with a fixed parameter is basically the same thing as a consumer.

Reply:
@jdimeo while that is true, this is a part of micro-optimization. Basically, when you pass in a parameter, you can cache it for that loop and avoid things like finality and reassignment issues. Secondly, the `*with` patterns help us use Method References.

Reply:
Understood. In general, I'm a big fan of how your APIs are not super spartan and provide some "unnecessary" conveniences like these, so 👍